database/rrdhost.c

```diff
@@ -811,7 +837,8 @@ void rrdhost_free(RRDHOST *host) {
 
 void rrdhost_free_all(void) {
     rrd_wrlock();
-    while(localhost) rrdhost_free(localhost);
+    while(localhost->next) rrdhost_free(localhost->next);
+    rrdhost_free(localhost);
```

Comment:
What's the difference with `while(localhost) rrdhost_free(localhost);`? It looks the same to me.

Reply:
> What's the difference with `while(localhost) rrdhost_free(localhost);`? It looks the same to me.

I want child-hosts to be released before the localhost. This ensures that a child on multihost will flush out any metrics. If the parent shutdowns first it will kill the dbengine. 

This will also help with the checks that were added to 

```
rrdeng_store_metric_flush_current_page
```

Reply:
maybe add this as one line coment as it is not immediatelly obvious why you do it this way